holoviews/core/accessors.py

```diff
@@ -0,0 +1,463 @@
+"""
+Module for accessor objects for viewable HoloViews objects.
+"""
+from __future__ import absolute_import, unicode_literals
+
+from collections import OrderedDict
+
+import param
+
+from . import util
+from .pprint import PrettyPrinter
+
+
+class Apply(object):
+    """
+    Utility to apply a function or operation to all viewable elements
+    inside the object.
+    """
+
+    def __init__(self, obj, mode=None):
+        self._obj = obj
+
+    def __call__(self, function, streams=[], link_inputs=True, dynamic=None, **kwargs):
+        """Applies a function to all (Nd)Overlay or Element objects.
+
+        Any keyword arguments are passed through to the function. If
+        keyword arguments are instance parameters, or streams are
+        supplied the returned object will dynamically update in
+        response to changes in those objects.
+
+        Args:
+            function: A callable function
+                The function will be passed the return value of the
+                DynamicMap as the first argument and any supplied
+                stream values or keywords as additional keyword
+                arguments.
+            streams (list, optional): A list of Stream objects
+                The Stream objects can dynamically supply values which
+                will be passed to the function as keywords.
+            link_inputs (bool, optional): Whether to link the inputs
+                Determines whether Streams and Links attached to
+                original object will be inherited.
+            dynamic (bool, optional): Whether to make object dynamic
+                By default object is made dynamic if streams are
+                supplied, an instance parameter is supplied as a
+                keyword argument, or the supplied function is a
+                parameterized method.
+            kwargs (dict, optional): Additional keyword arguments
+                Keyword arguments which will be supplied to the
+                function.
+
+        Returns:
+            A new object where the function was applied to all
+            contained (Nd)Overlay or Element objects.
+        """
+        from .dimension import ViewableElement
+        from .spaces import HoloMap, DynamicMap
+        from ..util import Dynamic
+
+        if isinstance(self._obj, DynamicMap) and dynamic == False:
+            samples = tuple(d.values for d in self._obj.kdims)
+            if not all(samples):
+                raise ValueError('Applying a function to a DynamicMap '
+                                 'and setting dynamic=False is only '
+                                 'possible if key dimensions define '
+                                 'a discrete parameter space.')
+            return HoloMap(self._obj[samples]).apply(
+                function, streams, link_inputs, dynamic, **kwargs)
+
+        if isinstance(function, util.basestring):
+            args = kwargs.pop('_args', ())
```

Comment:
Very minor suggestion, I might recommend renaming it to `_function_args` which makes it a little clearer what it is...

Reply:
Sounds good.

Reply:
I might go with ``_method_args`` as this is specifically meant for the ``apply.opts``, ``apply.select``, etc. accessors.